### ISSUE TITLE
Expose verbose error messages from Docker API

### ISSUE THREAD

**Issue by SKoschnicke** [denibertovic/docker-hs#45]
Currently, only the HTTP status code and the standard HTTP description for this code is returned on API errors. But the Docker API also always returns some JSON with a `message` attribute holding a more verbose description of the error which occurred. This description would be helpful for debugging and should also be returned in the `DockerError` structure.

**Comment by denibertovic** [comment]
@SKoschnicke I agree that this would be very useful. Do you want to try and take a stab at this?

**Comment by SKoschnicke** [comment]
Yes, but I'm not sure if I find time for this in the next days. If someone wants to work on this before me, he/she could comment here and start with it. Otherwise, I will add a notice as soon as I get to it.

**Comment by denibertovic** [comment]
That is fine, I'm a bit swamped currently as well. But lets coordinate here.
